### PR TITLE
feat(Vertex AI Gemini) — listener support (streaming and non streaming)

### LINKED ISSUE #1758: Vertex AI Gemini: add ChatModelListener support
Add `ChatModelListener` support to `VertexAiGeminiChatModel` and `VertexAiGeminiStreamingChatModel`.
See `OpenAiChatModel` and `OpenAiStreamingChatModel` as an example.

### PR DESCRIPTION
**Pull request by glaforge**
Fixes #1758

## Change

Add listener support to both Vertex AI Gemini streaming and non streaming models.

## General checklist

- [X] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

### CODE REVIEW

File: langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModel.java [hunk 1]
```diff
@@ -217,6 +222,7 @@ public VertexAiGeminiChatModel(GenerativeModel generativeModel,
         this.allowedFunctionNames = Collections.emptyList();
         this.logRequests = false;
         this.logResponses = false;
+        this.listeners = Collections.emptyList();
```
Comment (langchain4j) [inline langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModel.java:225]: Is it intended that there is no way to setup listeners via 2 remaining ctors?

File: langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModel.java [hunk 2]
```diff
@@ -188,6 +196,7 @@ public VertexAiGeminiStreamingChatModel(GenerativeModel generativeModel,
         this.allowedFunctionNames = Collections.emptyList();
         this.logRequests = false;
         this.logResponses = false;
+        this.listeners = Collections.emptyList();
```
Comment (langchain4j) [inline langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModel.java:199]: same here

**Review by ghost — COMMENTED** [review]
@glaforge thank you!

**Review by ghost — APPROVED** [review]
@glaforge thank you!

**Pull request merged**